src/main/java/runhistoryplus/patches/ImprovableCardRunHistoryPatch.java

```diff
@@ -0,0 +1,112 @@
+    private static final ArrayList<String> improvableCards = new ArrayList<String>() {{
+        add(GeneticAlgorithm.ID);
+        add(RitualDagger.ID);
+    }};
+    private static final String[] TEXT = CardCrawlGame.languagePack.getUIString("RunHistoryPlus:ImprovableCards").TEXT;
+
+    @SpirePatch(clz = CardCrawlGame.class, method = SpirePatch.CONSTRUCTOR)
+    public static class ImprovableCardsField {
+        @SpireRawPatch
+        public static void addImprovableCardsField(CtBehavior ctBehavior) throws NotFoundException, CannotCompileException {
+            CtClass runData = ctBehavior.getDeclaringClass().getClassPool().get(RunData.class.getName());
+            String fieldSource = "public java.util.Map improvable_cards;";
+            CtField field = CtField.make(fieldSource, runData);
+            runData.addField(field);
+        }
+    }
+
+    @SpirePatch(clz = Metrics.class, method = "gatherAllData")
+    public static class GatherAllDataPatch {
+        @SpirePostfixPatch
+        public static void gatherAllDataPatch(Metrics __instance, boolean death, boolean trueVictor, MonsterGroup monsters) {
+            Map<String, List<Integer>> improvableCardsLog = new HashMap<>();
+            for (final AbstractCard card: AbstractDungeon.player.masterDeck.group) {
+                if (improvableCards.contains(card.cardID)) {
+                    String metricID = card.getMetricID();
+                    if (!improvableCardsLog.containsKey(metricID)) {
+                        improvableCardsLog.put(metricID, new ArrayList<>());
+                    }
+                    improvableCardsLog.get(metricID).add(card.misc);
+                }
+            }
+            ReflectionHacks.privateMethod(Metrics.class, "addData", Object.class, Object.class)
+                    .invoke(__instance, "improvable_cards", improvableCardsLog);
+        }
+    }
+
+    @SpirePatch(clz = RunHistoryScreen.class, method = "update")
+    public static class RunHistoryScreenUpdatePatch {
+        @SpireInsertPatch(locator = Locator.class, localvars = "addMe")
+        public static void updateCardsWithFinalValues(RunHistoryScreen __instance, RunData ___viewedRun, TinyCard addMe) throws NoSuchFieldException, IllegalAccessException {
+            if (improvableCards.contains(addMe.card.cardID)) {
+                Field improvableCardsField = ___viewedRun.getClass().getField("improvable_cards");
+                Map<String, List<Double>> improvableCardsLog = (Map<String, List<Double>>)improvableCardsField.get(___viewedRun);
+                String metricID = addMe.card.getMetricID();
+                if (improvableCardsLog != null && improvableCardsLog.containsKey(metricID)) {
+                    Integer value = Collections.max(improvableCardsLog.get(metricID)).intValue();
+                    addMe.card.baseDamage = value;
+                    addMe.card.baseBlock = value;
+                    if (!addMe.card.rawDescription.contains(TEXT[0])) {
```

Comment:
Instead of patching the return value of `cardForName` I added a postfix patch to `reloadCards`, that loops over the cards and alters the description. Let me know if this works for you as well or if I should change it according to your proposal.

Reply:
Sounds good, that should work out roughly the same in terms of simplifying things and only running this operation once per run that gets looked at.